### ISSUE TITLE
[BUG] conda install 

### ISSUE THREAD

**Issue by andrea-bistacchi** [Loop3D/LoopStructural#115]
### Describe your issue

Hello, I installed in a conda environment with: `conda install -c loop3d loopstructural`, but then when I try to import LoopStructural I get an error saying `ModuleNotFoundError: No module named 'LoopProjectFile'`.

I am on Windows 10, Ananconda last version, Python 3.8.13 (last 3.8).

Thanks!

### Minimal reproducing code example

```python
import LoopStructural as loop3d
```

### Error message

```shell
Traceback (most recent call last):
  File "C:\Users\bistek\anaconda3\envs\pzero\lib\site-packages\LoopStructural\modelling\input\project_file.py", line 4, in <module>
    from LoopProjectFile import ProjectFile
ModuleNotFoundError: No module named 'LoopProjectFile'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\bistek\anaconda3\envs\pzero\lib\site-packages\LoopStructural\__init__.py", line 21, in <module>
    from .modelling.core.geological_model import GeologicalModel
  File "C:\Users\bistek\anaconda3\envs\pzero\lib\site-packages\LoopStructural\modelling\__init__.py", line 7, in <module>
    from LoopStructural.modelling.input.project_file import LoopProjectfileProcessor
  File "C:\Users\bistek\anaconda3\envs\pzero\lib\site-packages\LoopStructural\modelling\input\project_file.py", line 6, in <module>
    raise LoopImportError("LoopProjectFile cannot be imported")
LoopStructural.utils.exceptions.LoopImportError: LoopProjectFile cannot be imported
```

**Comment by andrea-bistacchi** [comment]
Hello, in some test installing 1.4.4 seems to solve the problem, so it could be a problem with 1.4.5.

Thanks!

**Comment by lachlangrose** [comment]
Hi Andrea,

LoopProjectFile is another project on loop3d anaconda channel. You could try `conda install -c conda-forge -c loop3 loopprojectfile`. This should get you going.

I will make this an optional dependency as its not actually required for loopstructural.

**Comment by andrea-bistacchi** [comment]
Hello Lachlan, and thanks very much for quick answer!

But it still do not work.

Apparently, I successfully installed `loopprojectfile` with `conda install -c loop3d loopprojectfile`, but when I try to import `loopstructural` I get a slightly different error.

Yesterday the error was `No module named 'LoopProjectFile'`.

Now it finds the 'LoopProjectFile' module, but cannot import 'ProjectFile'.
```
(pzero) D:\_REPOS\PZero>python

Python 3.8.13 (default, Mar 28 2022, 06:59:08) [MSC v.1916 64 bit (AMD64)] :: Anaconda, Inc. on win32
Type "help", "copyright", "credits" or "license" for more information.

>>> import LoopStructural as loop3d

Traceback (most recent call last):
  File "C:\Users\bistek\anaconda3\envs\pzero\lib\site-packages\LoopStructural\modelling\input\project_file.py", line 4, in <module>
    from LoopProjectFile import ProjectFile
ImportError: cannot import name 'ProjectFile' from 'LoopProjectFile' (C:\Users\bistek\anaconda3\envs\pzero\lib\site-packages\LoopProjectFile\__init__.py)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\bistek\anaconda3\envs\pzero\lib\site-packages\LoopStructural\__init__.py", line 21, in <module>
    from .modelling.core.geological_model import GeologicalModel
  File "C:\Users\bistek\anaconda3\envs\pzero\lib\site-packages\LoopStructural\modelling\__init__.py", line 7, in <module>
    from LoopStructural.modelling.input.project_file import LoopProjectfileProcessor
  File "C:\Users\bistek\anaconda3\envs\pzero\lib\site-packages\LoopStructural\modelling\input\project_file.py", line 6, in <module>
    raise LoopImportError("LoopProjectFile cannot be imported")
LoopStructural.utils.exceptions.LoopImportError: LoopProjectFile cannot be imported
```

**Comment by andrea-bistacchi** [comment]
Updates:

1) I also tried adding more modules, but I always get the same error as above:
```
conda install -c conda-forge -c loop3d loopstructural map2loop loopprojectfile mylibrary
```

2) To try being able to work while the problems above get fixed, I tried installing a previous version with:
```
conda install -c conda-forge -c loop3d loopstructural==1.4.4
conda install -c conda-forge -c loop3d loopstructural==1.4.3
conda install -c conda-forge -c loop3d loopstructural==1.4.2
```
But no one of these work and I always get:
```
Collecting package metadata (current_repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Collecting package metadata (repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.

PackagesNotFoundError: The following packages are not available from current channels:

  - loopstructural==1.4.3

Current channels:

  - https://conda.anaconda.org/conda-forge/win-64
  - https://conda.anaconda.org/conda-forge/noarch
  - https://conda.anaconda.org/loop3d/win-64
  - https://conda.anaconda.org/loop3d/noarch
  - https://repo.anaconda.com/pkgs/main/win-64
  - https://repo.anaconda.com/pkgs/main/noarch
  - https://repo.anaconda.com/pkgs/r/win-64
  - https://repo.anaconda.com/pkgs/r/noarch
  - https://repo.anaconda.com/pkgs/msys2/win-64
  - https://repo.anaconda.com/pkgs/msys2/noarch

To search for alternate channels that may provide the conda package you're
looking for, navigate to

    https://anaconda.org

and use the search bar at the top of the page.
```

This makes sense for 1.4.4, since there is no win-64 package on https://anaconda.org/loop3d/loopstructural/files, but 1.4.3 and 1.4.2 are listed there.

3) Finally, I reverted to pip, with `pip install LoopStructural`. This time I get the following error:
```
(pzero) D:\_REPOS\PZero>python

Python 3.8.13 (default, Mar 28 2022, 06:59:08) [MSC v.1916 64 bit (AMD64)] :: Anaconda, Inc. on win32
Type "help", "copyright", "credits" or "license" for more information.

>>> import LoopStructural as loop3d

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\bistek\anaconda3\envs\pzero\lib\site-packages\LoopStructural\__init__.py", line 21, in <module>
    from .modelling.core.geological_model import GeologicalModel
  File "C:\Users\bistek\anaconda3\envs\pzero\lib\site-packages\LoopStructural\modelling\__init__.py", line 6, in <module>
    from LoopStructural.modelling.input.process_data import ProcessInputData
  File "C:\Users\bistek\anaconda3\envs\pzero\lib\site-packages\LoopStructural\modelling\input\process_data.py", line 4, in <module>
    from LoopStructural.utils import strike_dip_vector
  File "C:\Users\bistek\anaconda3\envs\pzero\lib\site-packages\LoopStructural\utils\__init__.py", line 6, in <module>
    from .map2loop import process_map2loop, build_model
  File "C:\Users\bistek\anaconda3\envs\pzero\lib\site-packages\LoopStructural\utils\map2loop.py", line 6, in <module>
    from scipy.stats import truncnorm
  File "C:\Users\bistek\anaconda3\envs\pzero\lib\site-packages\scipy\__init__.py", line 136, in <module>
    from . import _distributor_init
  File "C:\Users\bistek\anaconda3\envs\pzero\lib\site-packages\scipy\_distributor_init.py", line 59, in <module>
    WinDLL(os.path.abspath(filename))
  File "C:\Users\bistek\anaconda3\envs\pzero\lib\ctypes\__init__.py", line 373, in __init__
    self._handle = _dlopen(self._name, mode)
FileNotFoundError: Could not find module 'C:\Users\bistek\anaconda3\envs\pzero\lib\site-packages\scipy\.libs\libbanded5x.TNNMG3IXHGJK7NIBT5J6YPEO5XWTOQAJ.gfortran-win_amd64.dll' (or one of its dependencies). Try using the full path with constructor syntax.
```
The same applies using previous versions `pip install LoopStructural==1.4.4`, 1.4.3, and 1.4.2.

Basically, it appears we are no more able to install LoopStructural.

Thanks!

**Comment by lachlangrose** [comment]
@andrea-bistacchi I have updated LoopStructural for windows/linux and it should work now without the project file. Let me know if this works for you now?

**Comment by andrea-bistacchi** [comment]
Hello, now the error when doing `import LoopStructural as loop3d` is:
```
ERROR: 2022-04-26 14:21:48,965: __init__.py:14 -- Loop project file cannot be imported
WARNING: 2022-04-26 14:21:48,995: __init__.py:57 -- Can't import surfepy - to install "pip install surfe"
WARNING: 2022-04-26 14:21:48,995: __init__.py:59 -- Using experimental interpolators: P1Interpolator and P2Interpolator
ERROR: 2022-04-26 14:21:49,102: intrusion_frame_builder.py:20 -- skfmm not installed
pip install scikit-fmm
ERROR: 2022-04-26 14:21:49,105: intrusion_builder.py:21 -- GeostatPy not installed
pip install geostatspy
GSLIB
```

**Comment by andrea-bistacchi** [comment]
However, if I import just the `GeologicalModel()` class, I can run model interpolation with `interpolator_type `set to either 'PLI', 'FDI' or 'surfe', even if the surfe module is apparently not available.
```
from LoopStructural import GeologicalModel
model = GeologicalModel(origin, maximum)
model.create_and_add_foliation("strati_0", interpolator_type=interpolator_type , nelements=(dimensions[0] * dimensions[1] * dimensions[2]))
regular_grid = model.regular_grid(nsteps=dimensions, shuffle=False, rescale=False)
scalar_field = model.evaluate_feature_value("strati_0", regular_grid, scale=False)
```

So it looks like the surfepy module is not really required.

**Comment by lachlangrose** [comment]
Hi Andrea,

Geostatspy and scikit-fmm are a few new requirements for the intrusions module. They should be optional dependencies, I'll fix this up. Yeah, for some reason geostatspy isn't on conda, you have to install from pip.

Surfepy should be an optional dependency... loopstructural will work without surfepy but you can't use the rbf interpolator then. It should be possible to pip install surfe or conda install -c conda-forge -c loop3d surfe.

**Comment by lachlangrose** [comment]
I just had another look at your error message, does the code run as you expect with the error messages for scikit-fmm, geostatspy and surfe? It should be ok if you are not trying to use the modules that use these libraries...

**Comment by andrea-bistacchi** [comment]
Hello, as said above, if I import just the `GeologicalModel()` class, instead of the whole LoopStructural library, I still see the errors and warnings:
```
ERROR: 2022-04-27 11:56:14,848: __init__.py:14 -- Loop project file cannot be imported
WARNING: 2022-04-27 11:56:14,900: __init__.py:57 -- Can't import surfepy - to install "pip install surfe"
WARNING: 2022-04-27 11:56:14,900: __init__.py:59 -- Using experimental interpolators: P1Interpolator and P2Interpolator
ERROR: 2022-04-27 11:56:15,161: intrusion_frame_builder.py:20 -- skfmm not installed 
pip install scikit-fmm
ERROR: 2022-04-27 11:56:15,164: intrusion_builder.py:21 -- GeostatPy not installed 
pip install geostatspy
```
but this is a non-blocking error and I can run model interpolation with interpolator_type set to either 'PLI', 'FDI' or 'surfe', even if the surfe module is apparently not available.
```
from LoopStructural import GeologicalModel
model = GeologicalModel(origin, maximum)
model.create_and_add_foliation("strati_0", interpolator_type=interpolator_type, nelements=(dimensions[0] * dimensions[1] * dimensions[2]))
regular_grid = model.regular_grid(nsteps=dimensions, shuffle=False, rescale=False)
scalar_field = model.evaluate_feature_value("strati_0", regular_grid, scale=False)
```
so it looks like that:
- loop project file, geostatpy and scikit-fmm are not really needed by GeologicalModel() (i.e. using loop as a library in our application);
- surfe interpolation works, when selected with `interpolator_type = 'surfe'`, even if surfepy is not installed;
- we do not use the experimental interpolators (maybe in the future?) so I have not tested them.

Thanks!
